src/v/storage/log_manager.h

```diff
@@ -227,7 +228,15 @@ class log_manager {
     int64_t compaction_backlog() const;
 
 private:
-    using logs_type = absl::flat_hash_map<model::ntp, log_housekeeping_meta>;
+    using logs_type
+      = absl::flat_hash_map<model::ntp, std::unique_ptr<log_housekeeping_meta>>;
+    using compaction_list_type = boost::intrusive::list<
+      log_housekeeping_meta,
+      boost::intrusive::member_hook<
+        log_housekeeping_meta,
+        log_housekeeping_meta::intrusive_link_type,
+        &log_housekeeping_meta::link>,
+      boost::intrusive::constant_time_size<false>>;
```

Comment:
i think using the typedefs in `utils/intrusive_list_helpers.h` will simplify this typedef a bit

Reply:
here is an example from `batch_cache.h`

```c++
intrusive_list<range, &range::_hook> _lru;
```